lib/wallet/test/unit/Cardano/Wallet/Shelley/TransactionSpec.hs

```diff
@@ -3248,13 +3143,15 @@ instance Buildable Wallet' where
 
 instance Arbitrary Wallet' where
     arbitrary = oneof
-        [ Wallet' allShelleyPaymentKeyCredentials
-            <$> (genWalletUTxO genShelleyVkAddr)
-            <*> (pure dummyShelleyChangeAddressGen)
-
-        , Wallet' allByronPaymentKeyCredentials
-            <$> (genWalletUTxO genByronVkAddr)
-            <*> (pure dummyByronChangeAddressGen)
+        [ genWalletUTxO genShelleyVkAddr <&> \utxo -> Wallet'
+            (allShelleyPaymentKeyCredentials (txWitnessTagFor @ShelleyKey))
```

Comment:
1. Not sure I like the `<&>`
2. Passing anything but `(txWitnessTagFor @ShelleyKey))` to `allShelleyPaymentCredentials` wouldn't make sense. We might as well apply it in the definition just below instead.